### PR TITLE
[#566] Add helm packing and publishing to local repo

### PR DESCRIPTION
**Pull request by serglupach**
it closes partially #566

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -477,7 +479,70 @@ EOL
                 }
             }
         }
-	}
+        stage('Package helm charts'){
```
Comment (alexander-semenets) [inline Jenkinsfile]: Can we do it in parallel with some other stage? Docker build for example...

File: Jenkinsfile [hunk 2]
```diff
@@ -477,7 +479,70 @@ EOL
...
+            steps{
+                dir ("${WORKSPACE}/deploy/helms") {
+                    script {
+                        chartNames = sh(returnStdout: true, script: 'ls').split()
+                        println (chartNames)
```
Comment (alexander-semenets) [inline Jenkinsfile]: Worth adding some description for the output if it's not abandoned debug which should be removed

File: Jenkinsfile [hunk 3]
```diff
@@ -477,7 +479,70 @@ EOL
...
+                               """
+                            }
+                        }
+                    }
+                }
+                dir ("${WORKSPACE}/legion-helm-charts") {
+                    script{
+                        if (params.StableRelease) {
+                            stage('Publish helm charts to Public repo'){
+                                if (params.UpdateVersionString){
```
Comment (alexander-semenets) [inline Jenkinsfile]: We should always set new version for release packages. UpdateVersionString parameter just says that version.py will be modified and uploaded to develop branch automatically.

File: Jenkinsfile [hunk 4]
```diff
@@ -477,7 +479,70 @@ EOL
...
+                                    git branch: "${params.HelmRepoGitBranch}", poll: false, url: "${params.HelmRepoGitUrl}"
+                                    //move packed charts to folder (where repo was checkouted)
+                                    for (chart in chartNames){
+                                        sh"""
+                                        cp ${WORKSPACE}/deploy/helms/${chart}-${Globals.buildVersion}.tgz ${WORKSPACE}/legion-helm-charts/
+                                        git add ${chart}-${Globals.buildVersion}.tgz
+                                        """
+                                    }
+                                    sh """
+                                    ls -alh
```
Comment (alexander-semenets) [inline Jenkinsfile]: Is this extra output really useful?

Comment (serglupach) [reply Jenkinsfile]: Not useful. Removed

File: Jenkinsfile (hunk 3 above)
Comment (serglupach) [reply Jenkinsfile]: Removed unneeded steps.

File: Jenkinsfile (hunk 2 above)
Comment (serglupach) [reply Jenkinsfile]: Added comment to this step.

File: Jenkinsfile (hunk 1 above)
Comment (serglupach) [reply Jenkinsfile]: Refactored. Moved stages to parallel stages

**Comment by serglupach** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
ok, to test

**Pull request merged**